src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/MessageQueueProcessorTest.java

```diff
@@ -139,14 +144,91 @@ public void run_should_consume_valid_message_so_they_are_not_reprocessed() throw
         // wait for message lock to expire
         Thread.sleep(testConfig.getMessageLockTimeoutInMs());
 
-        IMessage remainingMessage = receiveMessage();
-        assertThat(remainingMessage).as("check if non-null message was received").isNotNull();
-        assertThat(new String(remainingMessage.getBody())).isEqualTo(invalidMessageContent);
-
         // make sure that the valid message is not in the queue
         assertThat(receiveMessage()).as("check if subscription is empty").isNull();
     }
 
+    @Test
+    public void run_should_send_malformed_messages_to_dead_letter() throws Exception {
+        IMessage originalMessage = serviceBusFeeder.sendMessage("Invalid content");
+
+        messageQueueProcessor.run();
+
+        assertThat(receiveMessage()).as("check if subscription is empty").isNull();
+
+        IMessage deadLetterMessage = deadLetterQueueHelper.receiveMessage();
+        assertDeadLetterMessageWithMalformedContent(deadLetterMessage, originalMessage);
+
+        assertThat(deadLetterQueueHelper.receiveMessage())
+            .as("check if dead letter queue is empty")
+            .isNull();
+    }
+
+    @Test
+    public void run_should_send_well_formed_invalid_messages_to_dead_letter() throws Exception {
+        PrivateBetaRegistration invalidRegistration = getRegistrationWithInvalidEmail();
+        serviceBusFeeder.sendMessage(invalidRegistration);
+
+        messageQueueProcessor.run();
+
+        assertThat(receiveMessage()).as("check if subscription is empty").isNull();
+
+        IMessage deadLetterMessage = deadLetterQueueHelper.receiveMessage();
+
+        assertDeadLetterMessageIsForInvalidRegistration(
+            deadLetterMessage,
+            invalidRegistration,
+            "{\"emailAddress\":\"not a well-formed email address\"}");
+
+        assertThat(deadLetterQueueHelper.receiveMessage())
+            .as("check if dead letter queue is empty")
+            .isNull();
+    }
+
+    private void assertDeadLetterMessageIsForInvalidRegistration(
+        IMessage deadLetterMessage,
+        PrivateBetaRegistration invalidRegistration,
+        String validationErrorsString
+    ) throws JsonProcessingException {
+
+        assertThat(deadLetterMessage).as("check if dead letter message is present").isNotNull();
+
+        assertThat(bodyAsString(deadLetterMessage))
+            .as("dead letter message body")
+            .isEqualTo(objectMapper.writeValueAsString(invalidRegistration));
+
+        Map<String, String> expectedProperties = ImmutableMap.of(
+            "DeadLetterReason", "Invalid message",
+            "DeadLetterErrorDescription", "Message contains invalid data",
+            "ValidationErrors",  validationErrorsString
+        );
+
+        assertThat(deadLetterMessage.getProperties())
+            .as("dead letter message properties")
+            .isEqualTo(expectedProperties);
+    }
+
+    private void assertDeadLetterMessageWithMalformedContent(
```

Comment:
seems like a verb is missing

Reply:
Right, will change. How about `assertDeadLetterMessageIsForMalformedContent`?